Assets/UI/worldinput_CQUI.lua

```diff
@@ -665,22 +665,19 @@ end
 function OnMouseBuildingPlacementCancel( pInputStruct:table )
     -- print_debug("** Function Entry: OnMouseBuildingPlacementCancel (CQUI Hook)");
     if IsCancelAllowed() then
-        UI.SetInterfaceMode(InterfaceModeTypes.CITY_MANAGEMENT);
-        LuaEvents.CQUI_CityviewEnable();
         -- CQUI: Do not call ExitPlacementMode here
         -- ExitPlacementMode( true );
+        LuaEvents.CQUI_CityviewDisable();
     end
 end
 
 -- ===========================================================================
 function OnMouseDistrictPlacementCancel( pInputStruct:table )
     -- print_debug("** Function Entry: OnMouseDistrictPlacementCancel (CQUI Hook)");
     if IsCancelAllowed() then
-        LuaEvents.StoreHash(0);
-        UI.SetInterfaceMode(InterfaceModeTypes.CITY_MANAGEMENT);
-        LuaEvents.CQUI_CityviewEnable();
         -- CQUI: Do not call ExitPlacementMode here
         -- ExitPlacementMode( true );
+        LuaEvents.CQUI_CityviewDisable();
```

Comment:
Enable would work too since that is ultimately what the "CQUI_OnInterfaceModeChanged" function would decide to call. I chose to use disable (and make changes to the CQUI_CityviewDisableManager function) because I thought of it as "we are disabling District/Building Placement Mode, which is part of City View, so we are disabling part of City View." Upon reflection after receiving your comment though, this may be a bit confusing. The clearest option may be adding a new function titled something like "CQUI_PlacementModeDisable".

The main idea I had with regards to this was really to abstract away the interface change code from everything outside of CityPanel.lua. I thought it'd be better for CityPanel.lua to handle the interface changes as necessary and have everything outside of CityPanel.lua use functions such as this. What this line is ultimately doing though is just setting the interface mode back to CITY_MANAGEMENT.

The way it's setup in this PR does also prevent the Enable code from being run twice. If Enable was called instead of calling Disable (with my changes), then it would enable the City View and then set the interface mode to CITY_MANAGEMENT at the end, which would trigger the CQUI_OnInterfaceModeChanged function, which would then decide to call the Enable code again. There are other ways to fix this of course, I just simply chose this one.

I suppose a question to ask here would be: should the City View related interface mode changes be kept to just CityPanel.lua? If so, would a separate "CQUI_PlacementModeDisable" or similar function be better?

Reply:
I am impressed that you understand this code and not get confused by various lenses, colors, enable/disable events. etc. :)
Maybe it would help if you give a short description of how this mechanism works in general in the CityView?
Maybe then we could make it less complicated... I like things simple, with clear not confusing names :)

Reply:
I do feel like the code could maybe be simplified, yeah. I'm not completely happy with how it is right now, but I do think it's getting better at least.

I can try to give a short description of what's going on though:

As of this PR, the city view should be enabled and disabled with `LuaEvents.CQUI_CityviewEnable()`,  `LuaEvents.CQUI_CityviewDisable()`, and `LuaEvents.CQUI_CityviewDisableCurrentMode()` only. These will do all the necessary calls to the other files (like the city overview panel and production panel files) to enable/disable them as well. 

The first two of these LuaEvents will enable and disable the city view entirely. This will also disable the city view entirely even if in the District/Building placement mode beforehand.

The last one of these LuaEvents will, as the name says, disable the current city view mode. So if in the Placement mode, it will disable that mode and then restore the standard city view. Otherwise, it will disable the city view entirely.

I think the most confusing part of the code would be the `CQUI_OnInterfaceModeChanged` function. It's what determines when to open the Placement mode as well as when to enable/disable the city view, all based on how the interface mode changes. Though I think it's simpler now since I moved the `CQUI_DisplayGrowthTileIfValid` code out of the function to its own.

One small issue is that sometimes the `CQUI_OnInterfaceModeChanged` function calls `LuaEvents.CQUI_CityviewEnable()` and `LuaEvents.CQUI_CityviewDisable()` unnecessarily. For instance, if these events were called directly elsewhere, they will trigger an interface mode change, which will get picked up by the `CQUI_OnInterfaceModeChanged` function, which itself would then call the LuaEvents again. This would only cause one unnecessary repeat since the interface mode changed function is only called when the old and new modes are actually different, but it still feels bad that this is happening at all. I tried to fix this in an earlier version of the PR, but this caused other problems that I didn't notice at first. So I will leave it as is for now and resolve that in the future.